cylc/flow/task_pool.py

```diff
@@ -2169,6 +2172,7 @@ def force_trigger_tasks(
             if itask.state(TASK_STATUS_PREPARING, *TASK_STATUSES_ACTIVE):
                 LOG.warning(f"[{itask}] ignoring trigger - already active")
                 continue
+            self.merge_flows(itask, flow_nums)
```

Comment:
This is all that was needed to trigger an existing `n=0` task (e.g. queued) in a new flow. The rest of the PR is just making sure flow_nums change end up in the data store.